### DISCUSSION TITLE
The future of Gutenberg blocks in wp-graphql

### DISCUSSION THREAD

**Discussion by gu-stav** [wp-graphql/wp-graphql#1683] [Q&A] [answered]
Hi,
since Gutenberg becomes a more and more important and integral part of wordpress, I wanted to ask you about what you think about the future of blocks in relation to this plugin? Is it a possibility, that they will become part  of `wp-graphql` in a not so far future?

There is a [plugin](https://github.com/pristas-peter/wp-graphql-gutenberg) out there, which works but it's not very actively maintained and broken for the last version `wp-graphql`.

It would be great if there could be a reliable way to expose Gutenberg blocks via graphql, even though the problems mentioned by `wp-graphql-gutenberg` are quite hard to tackle.

Another option could be to bring the gutenberg plugin under the umbreall of the wp-graphql organisation, to keep the development going?

Any ideas or opinions?

**Comment by jasonbahl** [comment, marked as answer]
👋  Gutenberg is top of mind for me and GraphQL. It's not going to be a trivial task, but I'm thinking about it a lot. 

I've written my thoughts on the subject here: https://www.wpgraphql.com/2021/03/09/gutenberg-and-decoupled-applications/

**Comment by sahandnayebaziz** [comment]
Hey everyone, I just wanted to add that this sort of enhancement would be a huge productivity boost for me. I'm currently using the plugin, which seems to have some issues when it comes to correctly mapping metadata (for example, the attributes of images in gallery blocks) and so if I could vote for one enhancement to this excellent community of WP + GraphQL + Gatsby, it would be for official support for blocks and their metadata in the GraphQL API.
